backend/services/implementations/userService.ts

```diff
@@ -16,6 +17,41 @@ const getMongoUserByAuthId = async (authId: string): Promise<User> => {
   return user;
 };
 
+const getDateTime = (date: string): string =>
+  `${Math.floor(new Date(date).getTime() / 1000).toString(16)}0000000000000000`;
+
+interface MatchObject {
+  $match: {
+    _id: {
+      [key: string]: Types.ObjectId;
+    };
+  };
+}
+
+const buildMatchObject = (
+  startDate?: string,
+  endDate?: string,
+): MatchObject => {
+  const matchQuery: MatchObject = { $match: { _id: {} } };
+
+  if (startDate) {
+    // eslint-disable-next-line no-underscore-dangle
+    matchQuery.$match._id.$gt = Types.ObjectId(getDateTime(startDate));
+  }
+
+  if (endDate) {
+    // eslint-disable-next-line no-underscore-dangle
+    matchQuery.$match._id.$lt = Types.ObjectId(getDateTime(endDate));
+  }
```

Comment:
For a query with an open end (eg. look at all objects until now), is there any difference between saying `> startDate` vs `> startDate && <= Date.now`? in efficiency, style preference, etc.

Reply:
re: tests that makes sense, I've been doing that via manual test cases so far (in PR description)
re: comment, is that different from the intuitive usage? especially since we don't actually specify what a "beginning" or "end" date would be in the query

Reply:
Not sure I fully understand what you're saying, but I think what you have now is good. We can discuss it in today's meeting